A43-sdk-authorization.md

```diff
@@ -0,0 +1,364 @@
+A43: SDK authorization
+
+Author(s): [Ashitha Santhosh](https://github.com/ashithasantosh)
+Approver: 
+Status: Draft {Draft, In Review, Ready for Implementation, Implemented}
+Implemented in: <language, ...>
+Last updated: 2021-06-17
+Discussion at: (filled after thread exists)
+
+##Abstract
+
+gRPC is currently implementing an open-source standard authorization solution
+based on Envoy RBAC. This proposal discusses how this framework would be made 
+available to OSS gRPC users to perform per-RPC authorization checks.
+
+##Background
+
+gRPC authorization support which is currently under implementation, internally 
+implements RBAC Engines based on Envoy RBAC policies. RBAC policy provides 
+service-level and method-level access control for a service. Engines process 
+incoming RPC request attributes against policy configs and make a decision on 
+whether to allow or deny the request. The decision depends on the type of policy 
+(if the policy is an allowlist or denylist) and whether a matching policy was found.
+
+[RBAC filter]: https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/http/rbac/v3/rbac.proto
+
+The RBAC engine implementation is used by xDS API (i.e. Proxyless Service Mesh, 
+when Istio or Traffic Director sends Envoy RBAC policy to xDS enabled gRPC servers 
+for authorization) and SDK API (which makes authorization available to OSS gRPC users). 
+This document shares details on the latter.
+
+##Related Proposals:
+
+* [A41: xds RBAC Support][A41]
+
+##Proposal
+
+### User facing Authorization Policy
+
+gRPC SDK authorization policy is the user facing policy language that allows
+service owners and/or security admins to enable per-RPC authorization checks.
+
+SDK authorization policy guarantees stability and backward compatibility. So if
+a user writes a config, it will always be supported by gRPC unless the policy
+undergoes major versioning change like adding required fields. The policy is 
+extendable, we may add new fields in the future as new requirements are introduced, 
+we won’t be removing fields or updating semantics of existing fields.
+
+Following is the JSON schema of SDK Authorization Policy Version 1.0
+
+```json
```

Comment:
Can you reduce indentation to 2 spaces so it's easier to view the text?

Reply:
Sure. I reduced the indentation. It looks much cleaner now. Let me know if you would like me to reduce the line length further.

Reply:
Done.